cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/CreateSchedulesStage.java

```diff
@@ -103,5 +113,19 @@ public void process(ApplicationWithPrograms input) throws Exception {
 
   private void createSchedule(ProgramId programId, ScheduleSpecification scheduleSpec) throws SchedulerException {
     scheduler.schedule(programId, scheduleSpec.getProgram().getProgramType(), scheduleSpec.getSchedule());
+    addProgramSchedule(programId, scheduleSpec);
+  }
+
+  private void addProgramSchedule(ProgramId programId, ScheduleSpecification scheduleSpec) {
```

Comment:
So, now we have two stages: one to create old schedules, one to create new program schedules, and each one may create schedules of the other type. I find that rather confusing in terms of code structure. It appears that a single stage that deals with both types of schedules would have more clarity.